sdk/messaging/azservicebus/admin/admin_client_queue.go

```diff
@@ -271,39 +273,49 @@ type QueuePropertiesPager interface {
 	Err() error
 }
 
+type ListQueuesResult struct {
+	Items []*QueueItem
+}
+
 type ListQueuesResponse struct {
+	ListQueuesResult
 	RawResponse *http.Response
-	Value       []*QueueProperties
+}
+
+type QueueItem struct {
+	QueueName string
+	QueueProperties
 }
 
 // ListQueues lists queues.
-func (ac *AdminClient) ListQueues(options *ListQueuesOptions) QueuePropertiesPager {
-	var pageSize int
-	var skip int
+func (ac *Client) ListQueues(options *ListQueuesOptions) QueueItemPager {
```

Comment:
I agree with you here. However, that interface was taken from the guidelines:

https://azure.github.io/azure-sdk/golang_introduction.html#golang-pagination-pagers-interface-page

I didn't even think to question it because I remembered our discussion about making things easier to mock but I can remove it if we don't think the guideline is right.

Reply:
Removed in https://github.com/Azure/azure-sdk-for-go/pull/16141